packages/circuits-sdk/src/interfaces/ICommitmentWrapper.ts

```diff
@@ -33,6 +33,15 @@ export interface ICommitmentWrapper {
     commitment: Commitment,
     proof: ZKProof,
   ): ResultAsync<boolean, CircuitError>;
+
+  /**
+   * Initiates the loading of cryptographic resources (WASM and zkey files) needed for the zk operations.
+   * Method sets up asynchronous loaders to fetch and process these files and caches the results
+   * as promises within the class instance.
+   *
+   * This method should be called early in the application's lifecycle
+   */
+  preFetch(): ResultAsync<undefined, never>;
```

Comment:
Love the comment, but this suggests we may want to create a base interface, ICircomWrapper, with this method.